### ISSUE TITLE
how do I finish trimming after "manually" running some sorting jobs  

### ISSUE THREAD

**Issue by gdczoller** [marbl/canu#1001]
I am trying to finish the trimming step on a large genome in a cluster setting. Due to the very large disk footprint (>25TB) I had to run about half of the sorting jobs "manually", i.e. not via the canu submit script, but by running the 2-sort.jobSubmit-*.sh scripts in batches. Finally, I also ran the 3-index.sh "manually". I now have 2144 sorted trimming files, e.g. trimming/corr_test1.ovlStore.BUILDING/0001  as well as  the three files info, index and evalueLen from indexing. 
Then I restarted canu via the main submit script, but canu tries to sort again and complains that it cannot do that, e.g:
job trimming/corr_test1.ovlStore.BUILDING/0001 FAILED.
It seems canu is missing something and therefore does not know that sorting and indexing is already done. The bucketizing on the other hand it has acknowledged with "1-bucketize.success". Can I tell canu somehow, that sorting and indexing is done, e.g. with "2-sorting.success"? Or is there another way to tell canu to start from the sorted and indexed files? 
I am using canu version 1.6, as this was running for quite some time.

**Comment by skoren** [comment]
Assuming the re-indexing didn't change the previously created files, you can just rename corr_test1.ovlStore.BUILDING to corr_test1.ovlStore and it should continue. There is a 2-sorting.success but once the store exists, it doesn't get checked.

**Comment by gdczoller** [comment]
Thanks for the hints! After renaming it is now working in 3-overlapbasedtrimming. 
Coverage, according to the report: 
In gatekeeper store 'correction/corr_test1.gkpStore': 50.53 times
In gatekeeper store 'trimming/corr_test1.gkpStore': 34.46 times

A shy question: is that now the last step of the trimming phase? Can I somewhere find an overview of the sub-steps with the numbering as I see in the output? That would be very helpful when making decisions regarding disk space and run times on the cluster. Of course I understand, disk usage depends a lot on the genome size and how repetitive it is (and the parameter settings, I guess).

**Comment by brianwalenz** [comment]
I'm not sure we've documented the gory details anywhere.  The Canu paper has a wonderful flowchart, reproduced at http://canu.readthedocs.io/en/latest/pipeline.html

Running the ecoli at http://canu.readthedocs.io/en/latest/quick-start.html will do all the same steps, just not as many tasks per step (e.g., only one overlapper job is needed).

Peak disk usage is during correction.  Once that is finished - once you have correctedReads.fasta - you can be pretty aggressive cleaning up correction/.

**Comment by brianwalenz** [comment]
Keeping correctedReads lets you redo trimming, but there aren't really any parameters left to tweak during trimming.  So once you get trimmedReads, and they look sane - roughly the same amount of data as in correctedReads, roughly the same average read length - you can trash everything and start from trimmed reads (`canu ... -assemble -pacbio-corrected trimmedReads.fasta.gz`).

I'd suggest keeping the *.report from correction/trimming as a record of what was done.

**Comment by gdczoller** [comment]
May I ask, what output (regarding disk footprint) should I expect during/after the 3-overlapbasedtrimming step? Should I prepare for output in the TB range or in the GB range? Correction and trimming both produced output in the TB range so far, although most of the files of the correction step were eventually deleted. I am asking because we are a bit tight on the file system and we cannot expand it anymore.

**Comment by skoren** [comment]
I would expect since your trimming took TBs, the unitigging would be similar. However, you can remove both the correction/asm.ovlStore (once you have corrected reads, this happens automatically in canu version >=1.6) and the trimming/asm.ovlStore (once you have trimmed reads) to save space.

**Comment by skoren** [comment]
Idle, closing.
